### DISCUSSION TITLE
Extend the tools menu

### DISCUSSION THREAD

**Discussion by cjagannath** [codex-team/editor.js#1904] [Q&A]
Would be good to have an option to separate the tools menu(+) from the editor. This way we could scale the tools menu with more options and real estate and then add blocks using API.

**Comment by cjagannath** [comment]
Would help if the blocks API is exposed through the EditorJS instance. So we could do something like,

`const editor = new EditorJS({});
editor.api.blocks.insertNewBlock({});`

**Comment by neSpecc** [comment]
@cjagannath to the second reply: you already have this ability. We've add a note describing this at the top of the page: https://editorjs.io/api

**Comment by cjagannath** [comment]
Awesome. Then my only requirement would be to have an insertBlock API with block params as argument as mentioned in [Issue 715](https://github.com/codex-team/editor.js/issues/715). Is this planned in the near future?

**Comment by neSpecc** [comment]
@cjagannath let's discuss it at the #715 issue
